### PR TITLE
Fix SSR warnings with Next.js

### LINKED ISSUE #285: Warning with Next.js 10.2.0 and SSR
First, thank you for this great library! Unfortunately, I get a new warning after I upgraded to the latest Next.js version (v10.2.0) because of the server-side rendering in development mode: `Warning: Prop 'aria-describedby' did not match. Server: "DndDescribedBy-2" Client: "DndDescribedBy-1"` (full traceback below). It seems that the server-side ID increases each time I refresh the page (maybe because of the new stuff the Next.js team added to improve startup times).

I see that the ID is generated automatically with [`useUniqueId`](https://github.com/clauderic/dnd-kit/blob/f9ec28fed77778669f93cfecee159dba54db38b4/packages/core/src/components/DndContext/DndContext.tsx#L168) and without allowing to set it manually. A possible way to fix this I guess would be an optional `id` prop for the `DndContext` and pass it to `useUniqueId` to set the `draggableDescribedById` manually. What do you think?

Full traceback:
```
Warning: Prop `aria-describedby` did not match. Server: "DndDescribedBy-2" Client: "DndDescribedBy-1"
    at div
    at SortableItem (http://localhost:3000/_next/static/chunks/pages/templates/%5BtemplateName%5D.js?ts=1621372663479:308602:17)
    at SortableContext (http://localhost:3000/_next/static/chunks/pages/templates/%5BtemplateName%5D.js?ts=1621372663479:5647:3)
    at DndContext (http://localhost:3000/_next/static/chunks/pages/templates/%5BtemplateName%5D.js?ts=1621372663479:4498:3)
    at nav
    at http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:4155:66
    at List (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:25308:79)
    at div
    at http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:4155:66
    at Paper (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:30453:80)
    at SectionList (http://localhost:3000/_next/static/chunks/pages/templates/%5BtemplateName%5D.js?ts=1621372663479:307579:86)
    at div
    at http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:4155:66
    at Box (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:10205:99)
    at form
    at http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:4155:66
    at Box (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:10205:99)
    at FormProvider (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:125662:147)
    at StructureForm (http://localhost:3000/_next/static/chunks/pages/templates/%5BtemplateName%5D.js?ts=1621372663479:308867:23)
    at div
    at http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:4155:66
    at Box (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:10205:99)
    at StructurePanel (http://localhost:3000/_next/static/chunks/pages/templates/%5BtemplateName%5D.js?ts=1621372663479:309086:114)
    at div
    at http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:4155:66
    at Box (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:10205:99)
    at TransformerRegistry (http://localhost:3000/_next/static/chunks/pages/templates/%5BtemplateName%5D.js?ts=1621372663479:309831:23)
    at ModuleRegistry (http://localhost:3000/_next/static/chunks/pages/templates/%5BtemplateName%5D.js?ts=1621372663479:303510:23)
    at StructuredReportContent
    at TemplatePage
    at WithSuperJSON (http://localhost:3000/_next/static/chunks/pages/templates/%5BtemplateName%5D.js?ts=1621372663479:324597:42)
    at BlitzInnerRoot (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:1300:5)
    at main
    at http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:4155:66
    at Box (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:10205:99)
    at div
    at http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:4155:66
    at Container (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:15860:80)
    at div
    at http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:4155:66
    at Box (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:10205:99)
    at Layout (http://localhost:3000/_next/static/chunks/pages/templates/%5BtemplateName%5D.js?ts=1621372663479:311278:20)
    at Suspense
    at ErrorBoundary (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:125435:37)
    at InnerThemeProvider (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:50374:67)
    at ThemeProvider (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:54474:5)
    at ThemeProvider (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:50394:5)
    at SwitchableThemeProvider (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:63646:23)
    at LocalizationProvider (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:54316:5)
    at LocalizationProvider (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:63404:23)
    at App (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:66009:24)
    at AppWithStructuredReportTranslations (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:64385:28)
    at AppWithSiteTranslations (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:64235:28)
    at Provider (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:131913:20)
    at AppWithRedux (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:64099:32)
    at Hydrate (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:131181:23)
    at QueryClientProvider (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:131256:21)
    at BlitzProvider (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:1079:21)
    at BlitzOuterRoot (http://localhost:3000/_next/static/chunks/pages/_app.js?ts=1621372663479:1331:34)
    at ErrorBoundary (http://localhost:3000/_next/static/chunks/main.js?ts=1621372663479:142:47)
    at ReactDevOverlay (http://localhost:3000/_next/static/chunks/main.js?ts=1621372663479:245:23)
    at Container (http://localhost:3000/_next/static/chunks/main.js?ts=1621372663479:7839:5)
    at AppContainer (http://localhost:3000/_next/static/chunks/main.js?ts=1621372663479:8328:24)
    at Root (http://localhost:3000/_next/static/chunks/main.js?ts=1621372663479:8464:25)
```

### PR DESCRIPTION
**Pull request by medihack**
Fix the SSR warnings with the latest version of Next.js (v10.2.x).

Fixes #285

### CODE REVIEW

**Comment by medihack** [comment]
@clauderic If these changes are ok, should I do a minor bump (because of the added id prop) or a patch bump?

**Comment by medihack** [comment]
Removed some stuff I just tried out (seems I was too tired yesterday evening). Sorry for that, I think now it's ok.

File: packages/core/src/components/Accessibility/Accessibility.tsx [hunk 1]
```diff
@@ -22,6 +22,12 @@ export function Accessibility({
   const {announce, announcement} = useAnnouncement();
   const liveRegionId = useUniqueId(`DndLiveRegion`);
 
+  const [mounted, setMounted] = useState(false);
+
+  useEffect(() => {
+    setMounted(true);
+  }, []);
+
```
Comment (clauderic) [inline packages/core/src/components/Accessibility/Accessibility.tsx:30]: Why is this change needed?

Comment (medihack) [reply packages/core/src/components/Accessibility/Accessibility.tsx:30]: Hi! Without this, the client would render the `HiddenText` directly and Next.js would detect a mismatch between the stuff rendered on the server and the client ([that leads to the "Prop 'id' did not match" error](https://github.com/clauderic/dnd-kit/issues/285#issuecomment-844366815)). With the help of `useEffect` it is placed into the DOM after the first render, and so no mismatch occurs (that is why the [official `createPortal` of Next.js example does it that way](https://github.com/vercel/next.js/blob/canary/examples/with-portals/components/ClientOnlyPortal.js). I hope that makes things clear.

**Comment by medihack** [comment]
@clauderic Have you had some time to look at this PR? Can I do something to improve the PR? I also saw that [react-beautiful-dnd](https://github.com/atlassian/react-beautiful-dnd/blob/master/docs/api/reset-server-context.md) took another way to solve this problem (using a [resetServerContext function](https://github.com/atlassian/react-beautiful-dnd/blob/master/docs/api/reset-server-context.md) that must be called on the server). But IMHO an optional `id` for the `DndContext` is nicer and there would still be a problem with the `Accessibility` component.

**Comment by clauderic** [comment]
> I also saw that react-beautiful-dnd took another way to solve this problem (using a resetServerContext function that must be called on the server). But IMHO an optional id for the DndContext is nicer 

I agree.

Would you mind adding a [changeset](https://github.com/medihack/dnd-kit/new/fix-nextjs-ssr?filename=.changeset/nine-frogs-watch.md&value=---%0A%22%40dnd-kit%2Fcore%22%3A%20patch%0A---%0A%0AFix%20SSR%20warnings%20with%20Next.js%0A) to your PR that describes the changes you made?

**Comment by medihack** [comment]
Sure. I added a minor changeset as an optional `id` prop is added to the public `DndContext` API. Is this ok?

**Review by clauderic — APPROVED** [review]

**Comment by michaeljblum** [comment]
Hey @medihack - I saw your posts on react-beautiful-dnd's failure to play nice with more recent versions of Next and SSR (even with the resetServerContext workaround that seemed to work for earlier Next versions). Does dnd-kit work as expected with Next/SSR? 

Just asking before I fully commit. Library looks great, and is obviously supported, whereas r-b-dnd seems to be abandoned.

**Comment by medihack** [comment]
> 
> 
> Hey @medihack - I saw your posts on react-beautiful-dnd's failure to play nice with more recent versions of Next and SSR (even with the resetServerContext workaround that seemed to work for earlier Next versions). Does dnd-kit work as expected with Next/SSR?
> 
> Just asking before I fully commit. Library looks great, and is obviously supported, whereas r-b-dnd seems to be abandoned.
> 
> _Edit_: BTW, using {process.browser && } as a hacky way of getting it to work. Again, the resetServerContext fix didn't seem to do anything for me.

Yes, I completely switched over to this library and didn't regret it. And works perfectly with Next.js/Blitz.js/SSR.

**Pull request merged**